CUDA.cmake

```diff
@@ -37,6 +37,7 @@ endif()
 
 if(CUTLASS_NATIVE_CUDA)
 
+  set(CMAKE_CUDA_HOST_COMPILER ${CMAKE_CXX_COMPILER})
```

Comment:
I've debated adding this for a while now. It seems reasonable enough. Robert Maynard spoke a bit about why this isn't the case [here](https://gitlab.kitware.com/cmake/cmake/-/issues/17471), but since we require CXX and we know it will be active before this point in the CMake configuration, I think it should be safe to do this.

However, please test that setting CUDAHOSTCXX environment variable and CMAKE_CUDA_HOST_COMPILER from command line will override this setting. If not, we'll need to add guards to allow those overrides.